grails-app/services/de/iteratec/osm/result/ApplicationDashboardService.groovy

```diff
@@ -118,4 +126,29 @@ class ApplicationDashboardService {
 
     }
 
+    List<Map> getAllActivePagesAndMetrics(Long jobGroupId) {
+        List<Map> recentMetrics = getRecentMetricsForJobGroup(jobGroupId).collect {
+            it.projectedProperties.pageName = Page.findById(it.pageId).name
+            return it.projectedProperties
+        }
+
+        getPagesOfActiveJobs(jobGroupId).each {
+            Page page = it
+            if (page.name != Page.UNDEFINED &&
+                    !recentMetrics.any { it.pageId == page.id }) {
+                recentMetrics.add(
+                        [
+                                'speedIndex'                : null,
+                                'docCompleteTimeInMillisecs': null,
+                                'pageId'                    : page.id,
+                                'pageName'                  : page.name,
+                                'fullyLoadedIncomingBytes'  : null
+
```

Comment:
maybe you can add something like "page as Map" directly. Null values do not need to be set either way.

Reply:
done.